### ISSUE TITLE
create_sub_parent() Issue creating sub partitions with ID->Time or Time->ID when time control column is type bigint

### ISSUE THREAD

**Issue by JGovreau** [pgpartman/pg_partman#357]
I have my time control column as type bigint and tried multiple combinations of sub-partitioning:

1. ID -> Time
2. Time -> ID
3. Time -> Time

Out of these only Time->Time succeeded. The others failed with the below error:

```
ERROR:  invalid input syntax for integer: "3 mons"
CONTEXT:  PL/pgSQL function partman.create_sub_parent(text,text,text,text,text,text[],integer,text,boolean,text,text,boolean,boolean,boolean) line 164 at IF
```

After converting the time column to type timestamp instead of bigint, I was able to get the sub partitions to create with no issues. So it seems like it might be an issue with using bigint type for a time-based sub partition.

Here's how I'm trying to set it up:

Partitioned Table
```
CREATE TABLE cust_partition01.data_time_id (
    id bigint NOT NULL,
    time bigint
) PARTITION BY RANGE (time);
```

Template Table
```
CREATE TABLE partman.template_cust_partition01_data (
    id bigint NOT NULL,
    time bigint
);

ALTER TABLE ONLY partman.template_cust_partition01_data ADD CONSTRAINT data_pkey PRIMARY KEY (id);
```

Parent Partition
```
SELECT partman.create_parent(
	p_parent_table := 'cust_partition01.data_time_id', 
	p_control := 'time',
	p_type := 'native',
	p_interval := 'quarterly',
	p_epoch := 'seconds',
	p_template_table := 'partman.template_cust_partition01_data'
);
```

Parent Table Description
```
                          Table "cust_partition01.data_time_id"
 Column |  Type  | Collation | Nullable | Default | Storage | Stats target | Description
--------+--------+-----------+----------+---------+---------+--------------+-------------
 id     | bigint |           | not null |         | plain   |              |
 time   | bigint |           |          |         | plain   |              |
Partition key: RANGE ("time")
Partitions: data_time_id_p2020q2 FOR VALUES FROM ('1585717200') TO ('1593579600'),
            data_time_id_p2020q3 FOR VALUES FROM ('1593579600') TO ('1601528400'),
            data_time_id_p2020q4 FOR VALUES FROM ('1601528400') TO ('1609480800'),
            data_time_id_p2021q1 FOR VALUES FROM ('1609480800') TO ('1617253200'),
            data_time_id_p2021q2 FOR VALUES FROM ('1617253200') TO ('1625115600'),
            data_time_id_p2021q3 FOR VALUES FROM ('1625115600') TO ('1633064400'),
            data_time_id_p2021q4 FOR VALUES FROM ('1633064400') TO ('1641016800'),
            data_time_id_p2022q1 FOR VALUES FROM ('1641016800') TO ('1648789200'),
            data_time_id_p2022q2 FOR VALUES FROM ('1648789200') TO ('1656651600')
```

Create Sub Parent
```
SELECT partman.create_sub_parent(
	p_top_parent := 'cust_partition01.data_time_id',
	p_control := 'id',
	p_type := 'native',
	p_interval := '2500000',
	p_native_check := 'yes'
);
```

Error Message
```
ERROR:  invalid input syntax for integer: "3 mons"
CONTEXT:  PL/pgSQL function partman.create_sub_parent(text,text,text,text,text,text[],integer,text,boolean,text,text,boolean,boolean,boolean) line 164 at IF
```

However, I'm able to do the sub partition as a time-based with no issues
```
SELECT partman.create_sub_parent(
	p_top_parent := 'cust_partition01.data_time_id',
	p_control := 'time',
	p_type := 'native',
	p_interval := 'monthly',
	p_native_check := 'yes',
	p_epoch := 'seconds'
);

 create_sub_parent
-------------------
 t
```

I'm not sure if this is a bug, intended behavior, or something I'm doing wrong.

**Comment by keithf4** [comment]
Nope, that would be a bug. Seems I didn't cover all combination cases involving epoch on the parent. Will try and work on a bug fix when I get a chance, but I don't have an ETA at this time.

Or if you're curious and want to try and work on a patch, the initial condition is here:

https://github.com/pgpartman/pg_partman/blob/master/sql/functions/create_sub_parent.sql#L143

Followed up with this ELIF condition further down:

https://github.com/pgpartman/pg_partman/blob/master/sql/functions/create_sub_parent.sql#L179

Problem is you found a combination that does not match the first condition, but does match the second, so it thinks the top level partition set is not epoch.

**Comment by JGovreau** [comment]
Thanks for the info Kieth!

**Comment by keithf4** [comment]
So, I think this should fix the issue, but if you'd be able to test this as well, I'd appreciate it. This should hopefully let it just fall through since there's no boundaries that have to match up if one level is epoch and the other is integer.
```
diff --git a/sql/functions/create_sub_parent.sql b/sql/functions/create_sub_parent.sql
index c4889bb..f725160 100644
--- a/sql/functions/create_sub_parent.sql
+++ b/sql/functions/create_sub_parent.sql
@@ -176,7 +176,7 @@ LOOP
             RAISE EXCEPTION 'Due to conflicting data boundaries between ISO weeks and any larger interval of time, pg_partman cannot support a sub-partition interval of weekly';
         END IF;
 
-    ELSIF v_control_parent_type = 'id' AND v_control_sub_type = 'id' THEN
+    ELSIF v_control_parent_type = 'id' AND v_control_sub_type = 'id' AND v_parent_epoch == 'none' AND p_epoch == 'none' THEN
         IF p_interval::bigint >= v_parent_interval::bigint THEN
             EXECUTE format('SELECT set_config(%L, %L, %L)', 'search_path', v_old_search_path, 'false');
             RAISE EXCEPTION 'Sub-partition interval cannot be greater than or equal to the given parent interval';
```

**Comment by keithf4** [comment]
A fix for this has been incorporated into v4.5.1. Please let me know if it works for you.

**Comment by JGovreau** [comment]
Thanks Keith!
